packages/web-forms/src/components/FormQuestion.vue

```diff
@@ -13,11 +13,56 @@ const isSelectNode = (n: AnyLeafNode) : n is SelectNode => n.nodeType === 'selec
 </script>
 
 <template>
-	<div class="flex flex-column gap-2">
+	<div :id="question.nodeId + '_container'" class="question-container" :class="{ invalid: question.validationState.violation?.valid === false}">
 		<InputText v-if="isStringNode(question)" :question="question" />
 
 		<SelectControl v-else-if="isSelectNode(question)" :question="question" />
 
 		<UnsupportedControl v-else :question="question" />
+		
+		<div class="validation-placeholder">
+			<span class="validation-message">
+				{{ question.validationState.violation?.message.asString }}
+			</span>
+		</div>
 	</div>
 </template>
+
+<style scoped lang="scss">
+
+.question-container {
+	display: flex;
+	flex-direction: column;
+	padding: 0.5rem 1rem 0 1rem;
+	scroll-margin-top: 60px;
+}
+
+:global(.p-panel-content .question-container) {
+	margin-left: -1rem;
```

Comment:
What's happening here? It seems like it's accommodating spacing for the form-wide validation background, specific to questions in a group (and/or repeat?).

Reply:
that's right.

Reply:
Feels like this could use a comment. In general, I think spacing adjustments would probably benefit from some context/explanation when they use negative values, and values which are small or otherwise not obvious fractional proportions.